src/NuGet.Services.AzureSearch/Registration/RegistrationClient.cs

```diff
@@ -0,0 +1,43 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Net;
+using System.Threading.Tasks;
+using NuGet.Protocol.Catalog;
+
+namespace NuGet.Protocol.Registration
+{
+    public class RegistrationClient : IRegistrationClient
+    {
+        private readonly ISimpleHttpClient _simpleHttpClient;
+
+        public RegistrationClient(ISimpleHttpClient simpleHttpClient)
+        {
+            _simpleHttpClient = simpleHttpClient ?? throw new ArgumentNullException(nameof(simpleHttpClient));
+        }
+
+        public async Task<RegistrationIndex> GetIndexAsync(string indexUrl)
+        {
+            var result = await _simpleHttpClient.DeserializeUrlAsync<RegistrationIndex>(indexUrl);
+            if (result.StatusCode == HttpStatusCode.NotFound)
+            {
```

Comment:
It's not clear this returns null if the index doesn't exist. I would either:

1. Throw like `GetPageAsync` or `GetLeafAsycn`
2. Rename `GetIndexAsync` to something like `GetIndexOrNullAsync`.